.github/workflows/macos.yml

```
@@ -0,0 +1,75 @@
+name: 🍏 macOS
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.head_ref }}-macos
+  cancel-in-progress: true
+
+jobs:
+  build_appleclang:
+    name: AppleClang
+    runs-on: macos-latest
+    if: github.event.pull_request.draft == false
+    env:
+      CXXFLAGS: "-Werror -Wno-error=pass-failed"
+      # For macOS, Ninja is slower than the default:
+      #CMAKE_GENERATOR: Ninja
+      SETUPTOOLS_USE_DISTUTILS: stdlib
+    steps:
+    - uses: actions/checkout@v3
+    - name: Brew Cache
+      uses: actions/cache@v3
+      # - once stored under a key, they become immutable (even if local cache path content changes)
+      # - for a refresh the key has to change, e.g., hash of a tracked file in the key
+      with:
+        path: |
+          /usr/local
+          /Users/runner/Library/Caches/Homebrew
+        key: brew-macos-appleclang-${{ hashFiles('.github/workflows/macos.yml') }}
+        restore-keys: |
+          brew-macos-appleclang-
+    - name: install dependencies
+      run: |
+        brew --cache
+        set +e
+        brew unlink gcc
+        brew update
+        brew install --overwrite python
+        brew install ccache
+        brew install fftw
+        brew install libomp
+        brew link --force libomp
+        brew install ninja
+        brew install open-mpi
+        brew install pkg-config
+        set -e
+        brew tap openpmd/openpmd
+        brew install openpmd-api
+    - name: CCache Cache
+      uses: actions/cache@v3
+      # - once stored under a key, they become immutable (even if local cache path content changes)
+      # - for a refresh the key has to change, e.g., hash of a tracked file in the key
+      with:
+        path: /Users/runner/Library/Caches/ccache
+        key: ccache-macos-appleclang-${{ hashFiles('.github/workflows/macos.yml') }}-${{ hashFiles('cmake/dependencies/AMReX.cmake') }}
+        restore-keys: |
+          ccache-macos-appleclang-${{ hashFiles('.github/workflows/macos.yml') }}-
+          ccache-macos-appleclang-
+    - name: build ImpactX
```

ImpactX?

---

Copied wildy from WarpX and ImpactX xD
```suggestion
    - name: build HiPACE++
```